### PR TITLE
Modification of the widget "Deadlines/Events forwarded to somoene else"

### PR DESCRIPTION
**Pull request by areksolek**
Modification of the widget "Deadlines/Events forwarded to somoene else"

### CODE REVIEW

File: layouts/basic/modules/Vtiger/dashboards/CreatedNotMineActivities.tpl [hunk 1]
```diff
@@ -5,20 +5,54 @@
 	{assign var=CURRENTUSERID value=$CURRENTUSER->getId()}
 	<div class="dashboardWidgetHeader">
 		<div class="d-flex flex-row flex-nowrap no-gutters justify-content-between">
-			{include file=\App\Layout::getTemplatePath('dashboards/WidgetHeaderTitle.tpl', $MODULE_NAME) CLASSNAME="col-md-10"}
-			{include file=\App\Layout::getTemplatePath('dashboards/WidgetHeaderButtons.tpl', $MODULE_NAME)}
+			{include file=\App\Layout::getTemplatePath('dashboards/WidgetHeaderTitle.tpl', $MODULE_NAME) CLASSNAME="col-md-6"}
+			<div class="d-inline-flex">
+				{if \App\Privilege::isPermitted('Calendar', 'CreateView')}
+					<a class="btn btn-sm btn-light" onclick="Vtiger_Header_Js.getInstance().quickCreateModule('Calendar');
+							return false;" aria-label="{\App\Language::translate('LBL_ADD_RECORD')}" href="#"
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/dashboards/CreatedNotMineActivities.tpl]: ??
Widget.js

File: layouts/basic/modules/Vtiger/dashboards/CreatedNotMineActivities.tpl [hunk 2]
```diff
@@ -8,8 +8,7 @@
 			{include file=\App\Layout::getTemplatePath('dashboards/WidgetHeaderTitle.tpl', $MODULE_NAME) CLASSNAME="col-md-6"}
 			<div class="d-inline-flex">
 				{if \App\Privilege::isPermitted('Calendar', 'CreateView')}
-					<a class="btn btn-sm btn-light" onclick="Vtiger_Header_Js.getInstance().quickCreateModule('Calendar');
-							return false;" aria-label="{\App\Language::translate('LBL_ADD_RECORD')}" href="#"
+					<a class="btn btn-sm btn-light js-quickCreate_btn" data-js="click" aria-label="{\App\Language::translate('LBL_ADD_RECORD')}" href="#"
```
Comment (rskrzypczak) [inline layouts/basic/modules/Vtiger/dashboards/CreatedNotMineActivities.tpl]: Remove href. 
Change <a> to <button>

File: public_html/layouts/basic/modules/Vtiger/resources/dashboards/Widget.js [hunk 3]
```diff
@@ -2402,23 +2403,31 @@ YetiForce_Widget_Js('YetiForce_CalendarActivities_Widget_Js', {}, {
 			}
 		})
 	},
+	addQuickCreateButton: function () {
+		this.getContainer().find('.js-quickCreate_btn').on('click', function () {
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Vtiger/resources/dashboards/Widget.js]: Add to function registerHeaderButtons() and remove this

File: public_html/layouts/basic/modules/Vtiger/resources/dashboards/Widget.js [hunk 4]
```diff
@@ -1454,6 +1454,9 @@ jQuery.Class('Vtiger_Widget_Js', {
 			a[0].click();
 			a.remove();
 		});
+		container.find('.js-widget-quick-create').on('click', function () {
+			Vtiger_Header_Js.getInstance().quickCreateModule('Calendar');
```
Comment (rskrzypczak) [inline public_html/layouts/basic/modules/Vtiger/resources/dashboards/Widget.js]: remove 'Calendar' and get this from container

File: modules/Vtiger/views/List.php [hunk 5]
```diff
@@ -42,11 +42,9 @@ public function getPageTitle(\App\Request $request)
 		$title = App\Language::translate($moduleName, $moduleName);
 		$title = $title . ' ' . App\Language::translate('LBL_VIEW_LIST', $moduleName);
 
-		if ($request->has('viewname')) {
+		if ($request->has('viewname') && !empty(CustomView_Record_Model::getAll($moduleName)[$request->getByType('viewname', 2)])) {
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/views/List.php]: duplicate code CustomView_Record_Model::getAll

File: modules/Vtiger/views/List.php [hunk 6]
```diff
@@ -55,12 +53,10 @@ public function getBreadcrumbTitle(\App\Request $request)
 	{
 		$moduleName = $request->getModule();
 		$title = \App\Language::translate('LBL_VIEW_LIST', $moduleName);
-		if ($request->has('viewname')) {
+		if ($request->has('viewname') && !empty(CustomView_Record_Model::getAll($moduleName)[$request->getByType('viewname', 2)])) {
```
Comment (mariuszkrzaczkowski) [inline modules/Vtiger/views/List.php]: duplicate code CustomView_Record_Model::getAll

**Pull request merged**